numba/tests/test_dispatcher.py

```diff
@@ -179,6 +181,55 @@ def foo(x):
         self.assertPreciseEqual(foo(1), 3)
         self.assertPreciseEqual(foo(1.5), 3)
 
+    def test_inspect_llvm(self):
+
+        # Create a jited function
+        @jit
+        def foo(explicit_arg1, explicit_arg2):
+            return explicit_arg1 + explicit_arg2
+
+        # Call it in a way to create 3 signatures
+        foo(1, 1)
+        foo(1.0, 1)
+        foo(1.0, 1.0)
+
+        # base call to get all llvm in a dict
+        llvms = foo.inspect_llvm()
+        self.assertEqual(len(llvms), 3)
+
+        # make sure the function name shows up in the llvm
+        for llvm_bc in llvms.values():
+            # Look for the function name
+            self.assertTrue("foo" in llvm_bc)
+
+            # Look for the argument names
+            self.assertTrue("explicit_arg1" in llvm_bc)
+            self.assertTrue("explicit_arg2" in llvm_bc)
+
+    def test_inspect_asm(self):
+        # Create a jited function
+        @jit
+        def foo(explicit_arg1, explicit_arg2):
+            return explicit_arg1 + explicit_arg2
+
+        # Call it in a way to create 3 signatures
+        foo(1, 1)
+        foo(1.0, 1)
+        foo(1.0, 1.0)
+
+        # base call to get all llvm in a dict
+        asms = foo.inspect_asm()
+        self.assertEqual(len(asms), 3)
+
+        # make sure the function name shows up in the llvm
+        for asm in asms.values():
+            # Look for the function name
+            self.assertTrue("foo" in asm)
+
+            # Look for the argument names
+            self.assertTrue("explicit_arg1" in asm)
```

Comment:
Actually, you shouldn't find the argument names in the native assembler code (try it, using NUMBA_DUMP_ASSEMBLY :-)). Althought that might depend on the platform...